opensearch-operator/pkg/builders/cluster.go

```diff
@@ -972,3 +984,80 @@ func DataNodesCount(ctx context.Context, k8sClient client.Client, cr *opsterv1.O
 	}
 	return count
 }
+
+func NewServiceMonitor(cr *opsterv1.OpenSearchCluster) *v1.ServiceMonitor {
+
+	labels := map[string]string{
+		ClusterLabel: cr.Name,
+	}
+	selector := metav1.LabelSelector{
+		MatchLabels: labels,
+	}
+
+	namespaceSelector := v1.NamespaceSelector{
+		Any:        false,
+		MatchNames: []string{cr.Namespace},
+	}
+
+	if cr.Spec.General.Monitoring.ScrapInterval == "" {
+		cr.Spec.General.Monitoring.ScrapInterval = "30s"
+	}
+	user := v1.BasicAuth{}
+
+	monitorUser := cr.Spec.General.Monitoring.MonitoringUserSecret
+	if monitorUser == "" {
+		monitorUser = "admin"
+		user = v1.BasicAuth{
+			Username: corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: cr.Name + monitorUser + "-password"},
```

Comment:
Writing it like this is confusing as it is unclear where this secret is created.
Suggestion: Define variable `opensearchAdminSecret = cr.Name + "admin-secret` with a comment `// Use admin credentials if no separate monitoring user was defined` and use that.